### PR TITLE
core/state: remove unused error from prefetcher trie method

### PR DESCRIPTION
**Pull request by jwasinger**

### CODE REVIEW

File: core/state/state_object.go [hunk 1]
```diff
@@ -313,25 +313,19 @@ func (s *stateObject) updateTrie() (Trie, error) {
...
-		// Fetcher retrieval failed, something's very wrong, abort
-		s.db.setError(err)
-		return nil, err
-
-	case tr == nil:
+	tr := s.getPrefetchedTrie()
+	if tr == nil {
 		// Fetcher not running or empty trie, fallback to the database trie
-		tr, err = s.getTrie()
+		_, err := s.getTrie()
```
Comment (rjl493456442) [inline core/state/state_object.go:319]: it's wrong. 

originally `tr` will be assigned here and finally be returned by the function. 
now we just leave it as `nil` if the trie is not available in prefetcher

Comment (jwasinger) [reply core/state/state_object.go]: Yeah.  just pushed a fix for that.

File: core/state/state_object.go [hunk 2]
```diff
@@ -313,25 +313,18 @@ func (s *stateObject) updateTrie() (Trie, error) {
...
 	// Retrieve a pretecher populated trie, or fall back to the database
-	tr, err := s.getPrefetchedTrie()
-	switch {
-	case err != nil:
-		// Fetcher retrieval failed, something's very wrong, abort
-		s.db.setError(err)
-		return nil, err
-
-	case tr == nil:
+	tr := s.getPrefetchedTrie()
```
Comment (rjl493456442) [inline core/state/state_object.go:316]: maybe this way?

```go
	// Retrieve a pretecher populated trie, or fall back to the database
	tr := s.getPrefetchedTrie()
	if tr != nil {
		// Prefetcher returned a live trie, swap it out for the current one
		s.trie = tr
	} else {
		// Fetcher not running or empty trie, fallback to the database trie
		var err error
		tr, err = s.getTrie()
		if err != nil {
			s.db.setError(err)
			return nil, err
		}
	}
```
it's easier to understand with two code paths

File: core/state/state_object.go [hunk 3]
```diff
@@ -313,25 +313,21 @@ func (s *stateObject) updateTrie() (Trie, error) {
...
 		if err != nil {
 			s.db.setError(err)
 			return nil, err
 		}
-
-	default:
-		// Prefetcher returned a live trie, swap it out for the current one
-		s.trie = tr
 	}
+	s.trie = tr
```
Comment (rjl493456442) [inline core/state/state_object.go:329]: this assignment is duplicated, please remove it

**Review by rjl493456442 — APPROVED** [review]
one nitpick(or tiny issue), otherwise lgtm

File: core/state/state_object.go [hunk 4]
```diff
@@ -313,25 +313,21 @@ func (s *stateObject) updateTrie() (Trie, error) {
...
-	case err != nil:
-		// Fetcher retrieval failed, something's very wrong, abort
-		s.db.setError(err)
-		return nil, err
-
-	case tr == nil:
+	tr := s.getPrefetchedTrie()
+	if tr != nil {
+		// Prefetcher returned a live trie, swap it out for the current one
+		s.trie = tr
```
Comment (karalabe) [inline core/state/state_object.go:319]: This is redundant with Line 329

**Review by rjl493456442 — APPROVED** [review]
sgtm

**Review by frank0528 — APPROVED** [review]

**Review by fjl — APPROVED** [review]

**Pull request merged**